ruby-test-mode.el

```diff
@@ -69,49 +70,91 @@ Test Driven Development in Ruby."
   "The keymap used in `ruby-test-mode' buffers.")
 
 (defcustom ruby-test-file-name-extensions
-  '("builder" "erb" "haml" "rb" "rjs")
+  '("builder" "erb" "haml" "rb" "rjs" "rake" "slim")
   "*A list of filename extensions that trigger the loading of the
 minor mode."
   :type '(list)
   :group 'ruby-test)
 
 (defcustom ruby-test-implementation-filename-mapping
-  '(
-    ("\\(.*\\)\\(spec/controllers/\\)\\(.*\\)\\([^/]*\\)\\(_routing_spec\\)\\(\\.rb\\)$" "\\1config/routes.rb")
-    ("\\(.*\\)\\(spec/\\)\\(controllers\\|helpers\\|models\\)\\(.*\\)\\([^/]*\\)\\(_spec\\)\\(\\.rb\\)$" "\\1app/\\3\\4\\5\\7")
-    ("\\(.*\\)\\(test/\\)\\(controllers\\|helpers\\|models\\)\\(.*\\)\\([^/]*\\)\\(_test\\)\\(\\.rb\\)$" "\\1app/\\3\\4\\5\\7")
-    ("\\(.*\\)\\(spec/\\)\\(views\\)\\(.*\\)\\([^/]*\\)\\(_spec\\)\\(\\.rb\\)$" "\\1app/\\3\\4\\5")
-    ("\\(.*\\)\\(spec/\\)\\(lib/\\)\\(.*\\)\\([^/]*\\)\\(_spec\\)\\(\\.rb\\)$" "\\1\\3\\4\\5\\7")
-    ("\\(.*\\)\\(spec/\\)\\(.*\\)\\([^/]*\\)\\(_spec\\)\\(\\.rb\\)$" "\\1lib/\\3\\4\\6")
-    ("\\(.*\\)\\(test/\\)\\(unit/\\)\\(.*\\)\\([^/]*\\)\\(_test\\)\\(\\.rb\\)$" "\\1app/models/\\4\\5\\7" "\\1lib/\\4\\5\\7")
-    ("\\(.*\\)\\(test/\\)\\(functional/\\)\\(.*\\)\\([^/]*\\)\\(_test\\)\\(\\.rb\\)$" "\\1app/controllers/\\4\\5\\7")
-    ("\\(.*\\)\\(test/\\)\\(.*\\)\\([^/]*\\)\\(_test\\)\\(\\.rb\\)$" "\\1lib/\\3\\4\\6")
-    ("\\(.*\\)\\(_spec\\)\\(\\.rb\\)$" "\\1\\3")
-    ("\\(.*\\)\\(_test\\)\\(\\.rb\\)$" "\\1\\3"))
```

Comment:
I know those regex lines were already long. Could you keep them shorter? I think I would prefer something like this:
``` elisp
`((,(pcre-to-elisp "(.*)/spec/routing/routes_spec\\.rb$")
   "\\1/config/routes.rb")
  (,(pcre-to-elisp "(.*)/test/routing/routes_test\\.rb$")
   "\\1/config/routes.rb"))
```
What do you think?